src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php

```diff
@@ -386,6 +389,41 @@ function () {
         $this->assertFalse($form->get('field2')->isValid());
         $this->assertCount(1, $form->get('field2')->getErrors());
     }
+
+    public function testDisabledFormIsInvalidIfRootFormValidationFails()
+    {
+        $form = $this->formFactory
+            ->createBuilder(FormType::class, ['field1' => 0], [
+                'constraints' => new Callback(['callback' => static function ($data, ExecutionContextInterface $context) {
```

Comment:
Even though constraints validation linked to the root form are working correctly.